.github/workflows/semgrep.yml

```diff
@@ -0,0 +1,22 @@
+name: Semgrep
+on:
+  workflow_dispatch: {}
+  pull_request: {}
+  push:
+    branches:
+      - main
+  schedule:
+    # random HH:MM to avoid a load spike on GitHub Actions at 00:00
+    - cron: '35 11 * * *'
```

Comment:
Wouldn't we want to run this on PR's, not on merge to main or a schedule?

Reply:
It happens on both

Reply:
As written I think this runs on pushes to main, not on open PR's

Reply:
Ah wait I didn't see above, i'm silly